sparkle/platform/output.py

```diff
@@ -0,0 +1,298 @@
+                                                           objective,
+                                                           use_features)
+        self.configurator.scenario._set_paths(self.configurator.output_path)
+
+    def get_validation_data(self: ConfigurationOutput, instance_set: InstanceSet) -> \
+            tuple[ConfigurationResults, str]:
+        """Returns best config and ConfigurationResults for instance set."""
+        objective = self.configurator.scenario.sparkle_objective
+
+        # Retrieve found configuration
+        _, best_config = self.configurator.get_optimal_configuration(
+            self.solver, instance_set, objective.PerformanceMeasure)
+
+        # Retrieve validation results
+        validator = Validator(self.directory)
+        val_default = validator.get_validation_results(
+            self.solver, instance_set,
+            source_dir=self.directory, subdir="validation")
+        val_conf = validator.get_validation_results(
+            self.solver, instance_set, config=best_config,
+            source_dir=self.directory, subdir="validation")
+
+        results = []
+        # Form: 0: solver, 1: config, 2: set, 3: instance, 4: status,
+        # 5: quality, 6: runtime
+        for res in val_default:
+            # TODO: status to enum
+            results.append([res[3], res[4], res[5], res[6]])
+        results_default = ValResults(val_default[0][0], val_default[0][1],
+                                     val_default[0][2], results)
+
+        results = []
+        # Form: 0: solver, 1: config, 2: set, 3: instance, 4: status,
+        # 5: quality, 6: runtime
+        for res in val_conf:
+            # TODO: status to enum
+            results.append([res[3], res[4], res[5], res[6]])
+        results_conf = ValResults(val_conf[0][0], val_conf[0][1],
+                                  val_conf[0][2], results)
+
+        cutoff_time = self.configurator.scenario.cutoff_time
+        penalty = penalty_multiplier * \
+            self.configurator.scenario.cutoff_time
+        perf_par_conf = sgrfch.get_par_performance(val_default,
+                                                   cutoff_time,
+                                                   penalty,
+                                                   objective)
+        perf_par_def = sgrfch.get_par_performance(val_conf,
+                                                  cutoff_time,
+                                                  penalty,
+                                                  objective)
+        performance = ConfigurationPerformance(perf_par_conf, perf_par_def)
+
+        results = ConfigurationResults(performance, results_default, results_conf)
+        return results, best_config
+
+    def write_output(self: ConfigurationOutput) -> None:
+        """Write data into a json file."""
+        def serialize_configuration_results(cr: ConfigurationResults) -> None:
```

Comment:
Why a nested method? I would recommend two methods, where write_output becomes a very simple (pseudo code):
serialised_output = serialize_configuration_result()
json.write(serialised_output)

Reply:
Because we only access it from within this method so I though this might be cleaner then making them class methods, but if you prefer it that way I'll change it.